room_service.py

```diff
@@ -25,21 +35,50 @@ def __init__(self) -> None:
         pass
 
     def find_all(self) -> List[dict]:
-        cursor.execute(" SELECT * FROM rooms WHERE occupant = ''")
+        cursor.execute(" SELECT * FROM rooms WHERE numOccupants < maxOccupants ")
         return cursor.fetchall()
 
-    def reserve_room(self, room_id: str, student_name: str):
+    def reserve_room(self, room_id: str):
         """
             reserve room, by passing in room_id and student name
         """
-        cursor.execute(" UPDATE rooms SET occupant = :name WHERE roomNum = :room_id", {'name': student_name, 'room_id': room_id})
-        conn.commit()
+
+        # check if the room is available
+        cursor.execute(""" SELECT * FROM rooms WHERE roomID = ? """, (room_id,))
+        curr_room = cursor.fetchall()[0]
+
+        # if room is full (value of 1) then throw an err
+        # TODO: add an error throw statement
+        if curr_room[-1] == 1:
```

Comment:
extract `curr_room[-1]` into a variable like `isRoomFull` which would hold `True` if 1 or `False` if 0. (just to learn about ternary operators)

This is fine though but just variable names make it readable